### DISCUSSION TITLE
props intersectionObserver in SSR

### DISCUSSION THREAD

**Discussion by daehyun-thevc** [nuxt/nuxt#10312] [Questions (Nuxt 2)] [answered]
I want to receive a prop, instance of IntersectionObserver, like this:
```js
props: {
    intersectionObserver: { type: IntersectionObserver, required: true },
  },
```
However, error occurs like this: ```IntersectionObserver is not defined```
Because, in ssr, there is no definition of IntersectionObserver. Then, how can i check type of prop which is instance of IntersectionObserver without error?

**Comment by danielroe** [comment, marked as answer]
You can always do something like:

```js
props: {
  intersectionObserver: { type: process.client ? IntersectionObserver : Object, required: process.client },
},
```

But that does raise other questions about how you're working with this component on SSR.

**Reply by daehyun-thevc** [reply]
Thx, bro! it was very helpful! I think it would be no problem because I would use this intersectionObserver in mount step!
